yt_dlp/extractor/medici.py

```diff
@@ -1,67 +1,150 @@
+            'md5': '40f5e76cb701a97a6d7ba23b62c49990',
+            'info_dict': {
+                'id': '7857',
+                'ext': 'mp4',
+                'title': 'Carmen by Jiří Bubeníček after Roland Petit, music by Bizet, de Falla, Castelnuovo-Tedesco, and Bonolis',
+                'thumbnail': r're:^https?://medicitv-[abc]\.imgix\.net/movie/[^?]+\.jpg(?:\?[^?]+)?',
+                'description': 'md5:0f15a15611ed748020c769873e10a8bb',
+                'upload_date': '20240223',
+            },
+        },
+        {
+            'url': 'https://www.medici.tv/en/documentaries/la-sonnambula-liege-2023-documentaire',
+            'md5': '87ff198018ce79a34757ab0dd6f21080',
+            'info_dict': {
+                'id': '7513',
+                'ext': 'mp4',
+                'title': 'NEW: La Sonnambula',
+                'thumbnail': r're:^https?://medicitv-[abc]\.imgix\.net/movie/[^?]+\.jpg(?:\?[^?]+)?',
+                'description': 'md5:0caf9109a860fd50cd018df062a67f34',
+                'upload_date': '20231103',
+            },
+        },
+        {
+            'url': 'https://edu.medici.tv/en/masterclasses/yvonne-loriod-olivier-messiaen',
+            'md5': '5737b5b4d50a842605f5f7db6b76bce2',
+            'info_dict': {
+                'id': '3024',
+                'ext': 'mp4',
+                'title': 'Olivier Messiaen and Yvonne Loriod, pianists and teachers',
+                'thumbnail': r're:^https?://medicitv-[abc]\.imgix\.net/movie/[^?]+\.jpg(?:\?[^?]+)?',
+                'description': 'md5:aab948e2f7690214b5c28896c83f1fc1',
+                'upload_date': '20150223',
+            },
+        },
+        {
+            'url': 'https://www.medici.tv/en/jazz/makaya-mccraven-la-rochelle',
+            'md5': '4cc279a8b06609782747c8f50beea2b3',
+            'info_dict': {
+                'id': '7922',
+                'ext': 'mp4',
+                'title': 'NEW: Makaya McCraven in La Rochelle',
+                'thumbnail': r're:^https?://medicitv-[abc]\.imgix\.net/movie/[^?]+\.jpg(?:\?[^?]+)?',
+                'description': 'md5:b5a8aaeb6993d8ccb18bde8abb8aa8d2',
+                'upload_date': '20231228',
+            },
+        },
+    ]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
-        # Sets csrftoken cookie
         self._download_webpage(url, video_id)
+        cookies = self._get_cookies(url)
+
+        if "edu" in url:
+            request_url = f'https://api.medici.tv/edu-satie/edito/movie-file/{video_id}/'
+            source_url = 'https://edu.medici.tv'
+        else:
+            request_url = f'https://api.medici.tv/satie/edito/movie-file/{video_id}/'
+            source_url = 'https://www.medici.tv'
 
-        MEDICI_URL = 'http://www.medici.tv/'
+        mAuth_cookie = cookies.get('auth._token.mAuth')
+        if not mAuth_cookie:
+            self.to_screen('[auth] Can\'t find a token for authorization. Only previews may be downloaded.')
```

Comment:
Should this be a warning?

Reply:
I used self.to_screen because using self.report_warning stopped the test. Also, a few videos are available in full without a subscription, so preview only is not always the case. Is there a better way to deal with this?

Reply:
You can add `'expected_warnings': ['preview']` to test definition

> so preview only is not always the case.

Can we detect this?

Reply:
Now I updated so that it can detect the reason for previews. Also, I notice that the lifespan of the cookie 'auth._token.mAuth' is too short that the extractor almost always fail to fetch it. If anyone knows a better way aside the --cookies option, please let me know.

Reply:
is it a session cookie? session cookies cannot be extracted via `--cookies-from-browser` because they are not written to disk (or at least not written to the same location as regular cookies, in Chrome's case)

Reply:
If session cookies are the problem, ideally we would add support for login with credentials. But that's not required for this PR. You could raise with a login hint that mentions only `--cookies` will work / `--cookies-from-browser` doesn't work because the site uses session cookies.

Reply:
Yes, it's a session cookie, but stragely, it works fine with edu.medici.tv, whose cookie 'auth._token.mAuth' also seems to be session-based. Anyways, I already added `self.report_warning()` recommending users to use the `--cookies` option.

Reply:
Oops, I noticed a couple of misused double quotes. Now it's really finished.